libsignal-protocol-swift/Setup/Signal.swift

```diff
@@ -36,6 +36,25 @@ public final class Signal {
 }
 
 public extension Signal {
+    
+    static func publicKey(for privateKey: Data) throws -> Data {
+        
+        var pubKey: OpaquePointer? = nil
+        let result = withUnsafeMutablePointer(to: &pubKey) {
+            curve_generate_public_key($0, privateKey.signalBuffer)
```

Comment:
Should use `signal_buffer_free` to free the private key from memory in `defer` block.